app/services/katello/content_plugin_manager.rb

```diff
@@ -0,0 +1,48 @@
+module Katello
+  class ContentPluginManager
+    @content_types = {}
+    class << self
+      attr_reader :content_types
+      private :new
+
+      def def_field(*names)
+        class_eval do
+          names.each do |name|
+            define_method(name) do |*args|
+              args.empty? ? instance_variable_get("@#{name}") : instance_variable_set("@#{name}", *args)
+            end
+          end
+        end
+      end
+      # Plugin constructor
+      def register(id, &block)
+        content_type = new(id)
+        content_type.instance_eval(&block)
+        content_types[id] = content_type
+      end
+
+      def repository_types
+        content_types.keys.map(&:to_s)
+      end
+
+      def creatable_by_user?(repository_type)
+        find(repository_type).allow_creation_by_user
+      end
+
+      def find(repository_type)
+        content_types[repository_type.to_sym]
+      end
+    end
+
+    def_field :description, :allow_creation_by_user
```

Comment:
Well, we aren't piggy-backing on Foreman's plugin code here or inheriting it so I am not sure that's the best reason to use it. Even with your explanation, it's still confusing and I am not seeing the advantage. But I am probably missing something fundamental.